pkg/jsonschema/unique_constraint.go

```diff
@@ -0,0 +1,229 @@
+package jsonschema
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/metal-toolbox/governor-api/internal/models"
+	"github.com/santhosh-tekuri/jsonschema/v5"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+)
+
+// JSONSchemaUniqueConstraint is a JSON schema extension that provides a
+// "unique" property of type array
+var JSONSchemaUniqueConstraint = jsonschema.MustCompileString(
+	"https://governor/json-schemas/unique.json",
+	`{
+		"properties": {
+			"unique": {
+				"type": "array",
+				"items": {
+					"type": "string"
+				}
+			}
+		}
+	}`,
+)
+
+// UniqueConstraintSchema is the schema struct for the unique constraint JSON schema extension
+type UniqueConstraintSchema struct {
+	UniqueFieldTypesMap map[string]string
+	ERD                 *models.ExtensionResourceDefinition
+	ResourceID          *string
+	ctx                 context.Context
+	db                  boil.ContextExecutor
+}
+
+// UniqueConstraintSchema implements jsonschema.ExtSchema
+var _ jsonschema.ExtSchema = (*UniqueConstraintSchema)(nil)
+
+// Validate checks the uniqueness of the provided value against a database
+// to ensure the unique constraint is satisfied.
+func (s *UniqueConstraintSchema) Validate(_ jsonschema.ValidationContext, v interface{}) error {
+	// Skip validation if no database is provided
+	if s.db == nil {
+		return nil
+	}
+
+	// Skip validation if no constraint is provided
+	if len(s.UniqueFieldTypesMap) == 0 {
+		return nil
+	}
+
+	// Try to assert the provided value as a map, skip validation otherwise
+	mappedValue, ok := v.(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	qms := []qm.QueryMod{}
+
+	if s.ResourceID != nil {
+		qms = append(qms, qm.Where("id != ?", *s.ResourceID))
+	}
+
+	for k, value := range mappedValue {
+		// Convert the value to string
+		v := fmt.Sprint(value)
```

Comment:
why convert the value to a string here? should be it a type assertion? should it be for any type?

Reply:
the `Compile` function will ensure the types are not `object` or `array` (by using the `isValidType` function below). Here covert the value to string so I can use `"%s"` in the `Sprintf` statement below